pkg/policy/compiler.go

```diff
@@ -139,18 +165,35 @@ func (c *compiler) Compile(policy *v1alpha1.AuthorizationPolicy) (PolicyFunc, fi
 			})
 		}
 		for _, rule := range authorizations {
+			if rule.Match != nil {
+				// evaluate rule match condition
+				out, _, err := rule.Match.Eval(data)
+				if err != nil {
+					return nil, err
+				}
+				// evaluation result is nil, continue
+				if _, ok := out.(types.Null); ok {
+					continue
+				}
+				// try to convert to a check response
+				matched, err := utils.ConvertToNative[bool](out)
+				if err != nil {
+					return nil, err
+				}
+				// if condition is false, continue
+				if !matched {
+					continue
+				}
+			}
+
 			// evaluate the rule
-			out, _, err := rule.Eval(data)
+			rOut, _, err := rule.Response.Eval(data)
```

Comment:
```suggestion
			out, _, err := rule.Response.Eval(data)
```

Reply:
will not work as its a redeclaration of the out var from the match eval. I can either reassign it or group it in blocks